core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -285,7 +285,7 @@ private[spark] object Utils extends Logging {
         } else {
           // Restrict file permissions via chmod if available.
           // For Windows this step is ignored.
-          if (!isWindows && !chmod700(dir)) {
+          if (!isWindows && (conf==null || !isRunningInYarnContainer(conf)) && !chmod700(dir)) {
```

Comment:
if you do that then you don't have to change anything in `Worker`, because we're definitely not running in a YARN container there